axonserver/src/main/java/io/axoniq/axonserver/grpc/SubscriptionQueryRequestTarget.java

```diff
@@ -62,12 +61,14 @@
         this.context = context;
         this.subscriptionQueryInterceptors = subscriptionQueryInterceptors;
         this.unitOfWork = new DefaultPluginUnitOfWork(context, authentication);
-        this.errorHandler = e -> responseObserver.onError(GrpcExceptionBuilder.build(e));
+        this.errorHandler = e -> {
+            responseObserver.onError(GrpcExceptionBuilder.build(e));
+            unsubscribe();
```

Comment:
Ordering? (now you don't unsubscribe until onError has been handled.

Reply:
In this case, the order is not important.